dohq_artifactory/admin.py

```diff
@@ -363,7 +364,115 @@ def _create_json(self):
         return data_json
 
 
-class Repository(AdminObject):
+class GenericRepository(AdminObject):
+    @property
+    def path(self):
+        return self._artifactory.joinpath(self.name)
+
+    def _generate_query(self, package):
+        if self.packageType == Repository.DOCKER:
+            if ":" in package:
+                name, version = package.split(":")
+            else:
+                name = package
+                version = "*"
+
+            package = "/".join([name, version])
+
+            return {"name": "manifest.json", "path": {"$match": package}}
+
+        if self.packageType == Repository.PYPI:
+            operators = {
+                "<=": "$lte",
+                "<": "$lt",
+                ">=": "$gte",
+                ">": "$gt",
+                "==": "$eq",
+                "!=": "$ne",
+                "~=": "$match",
+            }
+            for symbol, operator in operators.items():
+                if symbol in package:
+                    name, version = package.split(symbol)
+
+                    return {
+                        "@pypi.name": {"$match": name},
+                        "@pypi.version": {operator: version},
+                    }
+
+            return {"@pypi.name": {"$match": package}}
+
+        if self.packageType == Repository.MAVEN:
+            package = package.replace(".", "/")
+
+        return {
+            "$or": [
+                {"name": {"$match": package}},
+                {"path": {"$match": package}},
+                {"@{}.name".format(self.packageType): {"$match": package}},
+                {"@build.name": {"$match": package}},
+                {"artifact.module.build.name": {"$match": package}},
+            ]
+        }
+
+    def _build_query(
+        self, terms=None, sort=None, include=None, limit=None, offset=None
+    ):
+        terms = terms or {}
+        terms["repo"] = {"$eq": self.name}
+
+        query = "items.find({terms})".format(terms=json.dumps(terms))
```

Comment:
Could we  use `aql` function from library instead of strings' operations? It makes the same, but with test (I don't remember exactly but hope with tests...). If `aql` is a object method feel free to change to static.

 https://github.com/devopshq/artifactory#artifactory-query-language 

```python
query = ['items.find', terms]
if include:
    query += ['.include', include ]
....
```

Reply:
Oh, I see. We use the function below in `search_raw`, but in a little weird way. Did you encounter an issue with using native python object in query?

Reply:
Sorry, I didn't know that such function present in the module. Replaced string with passing positional args.